### ISSUE TITLE
Updating AxisY Component?

### ISSUE THREAD

**Issue by kjell0w** [mhkeller/layercake#86]
Hey, I am filtering the data of a multi-line chart to hide some lines based on user input. I do it like this: 
```js
const {data} = getContext("LayerCake")
const originalData = $data

$: $data = originalData.filter(i => someCondition(i))
```
I get the desired behavior. Now I wonder if there is a way to update the AxisY component so that the scale adjusts to the currently filtered values?

**Comment by mhkeller** [comment]
It likely needs a keyed-each block, similar to here: https://github.com/mhkeller/layercake/issues/78#issuecomment-1136240949

I need to update the examples to include that...

**Comment by kjell0w** [comment]
Hey, thanks for your reply. I changed the each block in the AxisY component to `{#each tickVals as tick, i (tick)}`, but the Axis does not update the scale.

Before filtering
![image](https://user-images.githubusercontent.com/55231995/176210874-9213c1bf-e419-46ae-9dc9-e08b51ddeb61.png)

After filtering
![image](https://user-images.githubusercontent.com/55231995/176211030-6e44db3a-b6b6-4078-8a73-132d0ba678fb.png)

**Comment by kjell0w** [comment]
For sure! I copied the components from the multiline examples. I added an input above the chart and bound its value to a store, which is then set as the `ChartFilter` context. In `MultiLine.svelte` I filter the data based on this context. The component `AxisY.svelte` contains the keyed-each block.
Now, if you filter by `apple` or `banana` (the fruits with lower values on the Y-axis), the lines will be filtered correctly, but the scale of the Y-axis will not be decreased/updated.

https://svelte.dev/repl/8b8d4e09de64480ba3460662a8b3a1d1?version=3.48.0

**Comment by mhkeller** [comment]
Sorry for the delay I just saw this reply. 

Take a look at this setup: https://svelte.dev/repl/b52548c3baf048f9993e5588811ed867?version=3.48.0

Instead of filtering *inside* of the chart component, filter your data before it gets passed into the Layer Cake component. The key idea is that Layer Cake takes your data and measures the min and max of it. That's where the y axis is getting its values from. Because you were filtering inside of the multline component, the extent that the y-axis component was pulling from wasn't being updated.

In the example I made, I've reduced the complexity a bit and replaced `filterFruits` with a regular variable instead of a store. I'm filtering `dataLong` based on the value in the input and made that a reactive variable. Now, whenver the value of the input changes, your data itself is being filtered before it gets passed to the Layer Cake component. Whenever the data changes, the extent calculation happens and thus the y-axis values also update. Hope that helps clarify things.
